client/boot/project/wordpress-com.js

```diff
@@ -85,7 +87,11 @@ export function setupMiddlewares( currentUser, reduxStore ) {
 	// Render Layout only for non-isomorphic sections.
 	// Isomorphic sections will take care of rendering their Layout last themselves.
 	if ( ! document.getElementById( 'primary' ) ) {
-		renderLayout( reduxStore );
+		page( '*', function( context, next ) {
```

Comment:
Hmm... probably we don't. I need to think a little more about this part. The problem is that it was rendering the Layout synchronously, and I needed to make it async.

My understanding is that this code should render the Layout only if the server-side hasn't already done so. Maybe I should have the check for the `primary` element *inside* of the middleware, so it only renders if it needs to.

@ockham do have some insight here?